src/command/supergraph/compose/do_compose.rs

```diff
@@ -24,22 +26,45 @@ use crate::{
     RoverError, RoverErrorSuggestion, RoverOutput, RoverResult,
 };
 
-#[derive(Debug, Clone, Serialize, Parser)]
+#[derive(Debug, Serialize, Parser)]
 pub struct Compose {
+    #[clap(flatten)]
+    opts: SupergraphComposeOpts,
+}
+
+#[derive(Debug, Serialize, Parser)]
+pub struct SupergraphComposeOpts {
+    #[clap(flatten)]
+    pub plugin_opts: PluginOpts,
+
     /// The relative path to the supergraph configuration file. You can pass `-` to use stdin instead of a file.
     #[serde(skip_serializing)]
     #[arg(long = "config")]
     supergraph_yaml: FileDescriptorType,
 
-    #[clap(flatten)]
-    opts: PluginOpts,
+    /// A [`GraphRef`] that is accessible in Apollo Studio.
+    /// This is used to initialize your supergraph with the values contained in this variant.
+    ///
+    /// This is analogous to providing a supergraph.yaml file with references to your graph variant in studio.
+    ///
+    /// If used in conjunction with `--config`, the values presented in the supergraph.yaml will take precedence over these values.
+    #[arg(long = "graph-ref")]
+    graph_ref: Option<GraphRef>,
+
+    /// The version of Apollo Federation to use for composition
+    #[arg(long = "federation-version")]
+    federation_version: Option<FederationVersion>,
```

Comment:
We had to add more options to allow `graph-ref` but we can also add the option for `federation_version` to keep consistency with how `dev` works.